car_sim_lcm demo improvement

shensquared opened the pull request:
this further fixes the stability issue noted in #3258 and increases the target speed to 60mph.
- improves vehicle models (curb weight and exterior size matches the actual Prius)
- tunes contact and controller parameters
- corrects the gain block throttle and brake 20x magnification


+@liangfok for feature review, thanks. Also to be discussed: I initially added a PD steering controller on the right front wheel, which was essential under system 1 for stability. It felt hacky though, because from what I understand real cars have only one steering motor? Now that under system 2 the demo runs just fine without it, I'm leaving it out unless you feel otherwise?  

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/4023)
<!-- Reviewable:end -->

liangfok commented:
@shensquared, I finished my first pass of feature review. Overall looks good. Some questions. PTAL.

---

Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[drake/automotive/car_simulation.cc, line 153 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVji986XhcprdHKpUHQ:-KVji986XhcprdHKpUHR:b-eji41i) ([raw file](https://github.com/robotlocomotion/drake/blob/373274ca5218c6818812d294aba2da1b9c5fe951/drake/automotive/car_simulation.cc#L153)):*
> ```cpp
> 
>   // Sets up PD controllers for throttle and steering.
>   const double kpSteering = 100, kdSteering = 100, kThrottle = 250;
> ```

BTW. These changes are great. Note, however, that this method will be removed soon along with the rest of System 1.0. I see you've applied the changes to `car_sim_lcm.cc`. Thanks for doing that.

---

*[drake/automotive/car_simulation.cc, line 196 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjiRWb7QzRumhPJW-Y:-KVjiRWb7QzRumhPJW-Z:bkhuafx) ([raw file](https://github.com/robotlocomotion/drake/blob/373274ca5218c6818812d294aba2da1b9c5fe951/drake/automotive/car_simulation.cc#L196)):*
> ```cpp
>       map_driving_cmd_to_x_d(
>           tree->get_num_positions() + rigid_body->get_velocity_start_index(),
>           DrivingCommandIndices::kThrottle) = 1;
> ```

BTW, should this be `1 / radius of wheel`?

https://github.com/RobotLocomotion/drake/blob/master/drake/automotive/car_sim_lcm.cc#L214

---

*[drake/automotive/car_simulation.cc, line 201 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjiVsPQEOcV1VXAwc5:-KVjiVsPQEOcV1VXAwc6:b-dhncge) ([raw file](https://github.com/robotlocomotion/drake/blob/373274ca5218c6818812d294aba2da1b9c5fe951/drake/automotive/car_simulation.cc#L201)):*
> ```cpp
>       map_driving_cmd_to_x_d(
>           tree->get_num_positions() + rigid_body->get_velocity_start_index(),
>           DrivingCommandIndices::kBrake) = -1;
> ```

BTW, should this be `-1 / radius of wheel`?

https://github.com/RobotLocomotion/drake/blob/master/drake/automotive/car_sim_lcm.cc#L214

---

*[drake/automotive/steering_command_driver.py, line 34 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjia1O4NUSjwFUpz8f:-KVjia1O4NUSjwFUpz8g:b9526vg) ([raw file](https://github.com/robotlocomotion/drake/blob/373274ca5218c6818812d294aba2da1b9c5fe951/drake/automotive/steering_command_driver.py#L34)):*
> ```Python
> TURN_RIGHT_SIGN = -1.0
> 
> # target velocity 60mph, i.e. ~26.8224m/s
> ```

"target" should be "Target". Missing period at end of sentence.

---

*[drake/automotive/steering_command_driver.py, line 35 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjifP75Tul4LcbfrAp:-KVjifP75Tul4LcbfrAq:b-kjjpos) ([raw file](https://github.com/robotlocomotion/drake/blob/373274ca5218c6818812d294aba2da1b9c5fe951/drake/automotive/steering_command_driver.py#L35)):*
> ```Python
> 
> # target velocity 60mph, i.e. ~26.8224m/s
> MAX_VELOCITY = 26.8224 
> ```

Extra space at end of line.

---

*[drake/automotive/models/prius/prius.sdf, line 11 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjilXuug_jUaWhZjnr:-KVjilXuug_jUaWhZjns:b-8a8nrs) ([raw file](https://github.com/robotlocomotion/drake/blob/373274ca5218c6818812d294aba2da1b9c5fe951/drake/automotive/models/prius/prius.sdf#L11)):*
> ```
>       <inertial>
>        <pose>0 0 0 0 0 0</pose>
>         <mass>1360</mass>
> ```

Can you add a comment pointing the reader to an authoritative source for this value?

---

*[drake/automotive/models/prius/prius.sdf, line 13 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjit3wWrfsX7bjVq6t:-KVjit3wWrfsX7bjVq6u:b-igd38s) ([raw file](https://github.com/robotlocomotion/drake/blob/373274ca5218c6818812d294aba2da1b9c5fe951/drake/automotive/models/prius/prius.sdf#L13)):*
> ```
>         <mass>1360</mass>
>         <inertia>
>           <ixx>63.858358934</ixx>
> ```

How were these values derived? (Perhaps add a comment.)

---

*[drake/automotive/models/prius/prius.sdf, line 391 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjj-EOIrzSBx0xrd6T:-KVjj-EP0fOcORrkkj6A:b-lieuje) ([raw file](https://github.com/robotlocomotion/drake/blob/373274ca5218c6818812d294aba2da1b9c5fe951/drake/automotive/models/prius/prius.sdf#L391)):*
> ```
>       </visual>
>       <collision name='wheel'>
>            <laser_retro>0</laser_retro>
> ```

What is the purpose of `<laser_retro>0</laser_retro>`? Also, it seems to be indented too much.

---

*[drake/automotive/models/prius/prius.sdf, line 392 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjjuXjuvy9qW7B0jFx:-KVjjuXjuvy9qW7B0jFy:b-ahtapo) ([raw file](https://github.com/robotlocomotion/drake/blob/373274ca5218c6818812d294aba2da1b9c5fe951/drake/automotive/models/prius/prius.sdf#L392)):*
> ```
>       <collision name='wheel'>
>            <laser_retro>0</laser_retro>
>         <max_contacts>10</max_contacts>
> ```

Is this tag used by Drake? I searched for "max_contact" in `parser_sdf.cc` but didn't find it:

https://github.com/RobotLocomotion/drake/blob/master/drake/systems/plants/parser_sdf.cc

---

*[drake/automotive/models/prius/prius.sdf, line 407 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjkgKTxGRECuZe2mvr:-KVjkgKTxGRECuZe2mvs:bkbtvuu) ([raw file](https://github.com/robotlocomotion/drake/blob/373274ca5218c6818812d294aba2da1b9c5fe951/drake/automotive/models/prius/prius.sdf#L407)):*
> ```
>       <inertial>
>         <pose>0 0 0 0 0 0</pose>
>         <mass>54.2794</mass>
> ```

Do you have a citation for this value?

---

*[drake/automotive/models/prius/prius_with_lidar.sdf, line 391 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjkzc8RNcd1nb3FThT:-KVjkzc8RNcd1nb3FThU:b-h0s2dj) ([raw file](https://github.com/robotlocomotion/drake/blob/373274ca5218c6818812d294aba2da1b9c5fe951/drake/automotive/models/prius/prius_with_lidar.sdf#L391)):*
> ```
>       </visual>
>       <collision name='wheel'>
>                 <laser_retro>0</laser_retro>
> ```

Ditto these additions. (See comment in `prius.sdf`.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-:-KVjf3O0uYIFddayqW8B:bunlb5)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
@drake-jenkins-bot add to whitelist

david-german-tri commented:
@shensquared Apparently you aren't a member of the RobotLocomotion org yet.  I've sent you an invite.

liangfok commented:
>  I initially added a PD steering controller on the right front wheel, which was essential under system 1 for stability. It felt hacky though, because from what I understand real cars have only one steering motor? Now that under system 2 the demo runs just fine without it, I'm leaving it out unless you feel otherwise?

Yes, I would leave that out. We'll eventually want to increase the fidelity of the steering mechanism within the model. It will likely involve more sophisticated models and controllers.

---

Review status: all files reviewed at latest revision, 11 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-:-KVjpFNlHeCERrElTbBX:bpo85fj)*
<!-- Sent from Reviewable.io -->

shensquared commented:
Thanks for the comments @liangfok, I addressed them below. (btw, thanks for adding me to the team @david-german-tri)

---

Review status: 1 of 6 files reviewed at latest revision, 11 unresolved discussions.

---

*[drake/automotive/car_simulation.cc, line 153 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVji986XhcprdHKpUHQ:-KVl1xsJvnDfzhF57uVu:b-1vtvpy) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/car_simulation.cc#L153)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> BTW. These changes are great. Note, however, that this method will be removed soon along with the rest of System 1.0. I see you've applied the changes to `car_sim_lcm.cc`. Thanks for doing that.
</details>
no problem :)

---

*[drake/automotive/car_simulation.cc, line 196 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjiRWb7QzRumhPJW-Y:-KVko7WpKrMhH1SlT5Kv:bwufdjz) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/car_simulation.cc#L196)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> BTW, should this be `1 / radius of wheel`?
> 
> https://github.com/RobotLocomotion/drake/blob/master/drake/automotive/car_sim_lcm.cc#L214
</details>
right, done. 

---

*[drake/automotive/car_simulation.cc, line 201 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjiVsPQEOcV1VXAwc5:-KVkpMHXOusTJJWVZ9Xf:bnj7hl0) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/car_simulation.cc#L201)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> BTW, should this be `-1 / radius of wheel`?
> 
> https://github.com/RobotLocomotion/drake/blob/master/drake/automotive/car_sim_lcm.cc#L214
</details>
same as above

---

*[drake/automotive/steering_command_driver.py, line 34 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjia1O4NUSjwFUpz8f:-KVkpPwUv5Dfx1SJHtF5:bb7f2mf) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/steering_command_driver.py#L34)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> "target" should be "Target". Missing period at end of sentence.
</details>
done

---

*[drake/automotive/steering_command_driver.py, line 35 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjifP75Tul4LcbfrAp:-KVkpQb_lTWMTnQJSGRa:b-535kmx) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/steering_command_driver.py#L35)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Extra space at end of line.
</details>
done.  

---

*[drake/automotive/models/prius/prius.sdf, line 11 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjilXuug_jUaWhZjnr:-KVkprGnkbEy98RklVEI:b-7rl0jd) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/models/prius/prius.sdf#L11)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Can you add a comment pointing the reader to an authoritative source for this value?
</details>
done.

---

*[drake/automotive/models/prius/prius.sdf, line 13 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjit3wWrfsX7bjVq6t:-KVkpsiBbl3v3oX-ENKg:b-7rl0jd) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/models/prius/prius.sdf#L13)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> How were these values derived? (Perhaps add a comment.)
</details>
done.

---

*[drake/automotive/models/prius/prius.sdf, line 391 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjj-EOIrzSBx0xrd6T:-KVkptLT-akSPwJZP9ia:b-3e02ns) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/models/prius/prius.sdf#L391)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> What is the purpose of `<laser_retro>0</laser_retro>`? Also, it seems to be indented too much.
</details>
this tag was there for all the other three wheels. I verified it doesn't hurt the dynamics by `comepareRigidBodySystems` with the URDF model, so added it just for symmetry. It doesn't look like it's serving any purpose either? I removed them all now. 

---

*[drake/automotive/models/prius/prius.sdf, line 392 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjjuXjuvy9qW7B0jFx:-KVkqT-lji0PvzNB-WoE:brx2yk6) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/models/prius/prius.sdf#L392)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Is this tag used by Drake? I searched for "max_contact" in `parser_sdf.cc` but didn't find it:
> 
> https://github.com/RobotLocomotion/drake/blob/master/drake/systems/plants/parser_sdf.cc
</details>
same as above. 

---

*[drake/automotive/models/prius/prius.sdf, line 407 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjkgKTxGRECuZe2mvr:-KVkqbjgPcQOvzu1ect5:bthrltq) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/models/prius/prius.sdf#L407)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Do you have a citation for this value?
</details>
I haven't found authoritative resources online for the metal body specs. The body is not of regular shape so I could only guess its weight, 10 times the old mass felt right to me, so did just that, and the inertia is increased ten fold as well. 

I chatted with @Sproulx who put together the old mass and inertia values a long time ago, he couldn't remember where they originated from, but perhaps estimated using SolidWorks. He'll be out next week but could help verify the values again if necessary.

---

*[drake/automotive/models/prius/prius_with_lidar.sdf, line 391 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjkzc8RNcd1nb3FThT:-KVl2vg2Xu2nrIvD40Cl:bclm5g5) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/models/prius/prius_with_lidar.sdf#L391)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Ditto these additions. (See comment in `prius.sdf`.)
</details>
same changes as in `prius.sdf`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-:-KVl4Vpw433neH2S0pjZ:b21k1q1)*
<!-- Sent from Reviewable.io -->

liangfok commented:
@shensquared I finished my second round of review. Getting close!

---

Reviewed 5 of 5 files at r2.
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

*[drake/automotive/car_simulation.cc, line 196 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVlhmXx55n_MsoXus4X:-KVlhmXx55n_MsoXus4Y:b-93dxfd) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/car_simulation.cc#L196)):*
> ```cpp
>       map_driving_cmd_to_x_d(
>           tree->get_num_positions() + rigid_body->get_velocity_start_index(),
>           DrivingCommandIndices::kThrottle) = 1/0.323342;
> ```

BTW: nit: missing spaces before and after "/".

---

*[drake/automotive/car_simulation.cc, line 201 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVlhrf7ftANSEzMWxR3:-KVlhrf7ftANSEzMWxR4:b-93dxfd) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/car_simulation.cc#L201)):*
> ```cpp
>       map_driving_cmd_to_x_d(
>           tree->get_num_positions() + rigid_body->get_velocity_start_index(),
>           DrivingCommandIndices::kBrake) = -1/0.323342;
> ```

BTW: nit: missing spaces before and after "/".

---

*[drake/automotive/models/prius/prius.sdf, line 391 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjj-EOIrzSBx0xrd6T:-KVljOGdwS94dWbOdn2q:b-ucfknh) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/models/prius/prius.sdf#L391)):*
<details><summary><i>Previously, shensquared (Shen Shen) wrote…</i></summary>
> this tag was there for all the other three wheels. I verified it doesn't hurt the dynamics by `comepareRigidBodySystems` with the URDF model, so added it just for symmetry. It doesn't look like it's serving any purpose either? I removed them all now. 
</details>
Thanks!

---

*[drake/automotive/models/prius/prius.sdf, line 407 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjkgKTxGRECuZe2mvr:-KVlj_xWxVIUuyErwJSK:bfxi830) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/models/prius/prius.sdf#L407)):*
<details><summary><i>Previously, shensquared (Shen Shen) wrote…</i></summary>
> I haven't found authoritative resources online for the metal body specs. The body is not of regular shape so I could only guess its weight, 10 times the old mass felt right to me, so did just that, and the inertia is increased ten fold as well. 
> 
> I chatted with @Sproulx who put together the old mass and inertia values a long time ago, he couldn't remember where they originated from, but perhaps estimated using SolidWorks. He'll be out next week but could help verify the values again if necessary.
</details>
OK. Please add a `TODO` comment saying this mass value needs to be verified / fixed.

---

*[drake/automotive/models/prius/prius.sdf, line 11 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVlijrs2t39l3C8yYLx:-KVlijrs2t39l3C8yYLy:b-9zh6u) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L11)):*
> ```
>       <inertial>
>        <pose>0 0 0 0 0 0</pose>
>         <!-- curb weight 3,072 lbs. (~1400 kg) based on 
> ```

This should be a complete sentence with proper punctuation per style guide.

```
The 2016 Toyota Prius has a curb weight of 3,072 lbs. (~1,400 kg) based on the PDF
linked to below.

https://pressroom.toyota.com/releases/toyota+2015+prius+product+specs.download
```

---

*[drake/automotive/models/prius/prius.sdf, line 13 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVli9Xw1VnutUrxUIsm:-KVli9Xw1VnutUrxUIsn:bcfb7x6) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L13)):*
> ```
>         <!-- curb weight 3,072 lbs. (~1400 kg) based on 
>         https://pressroom.toyota.com/releases/toyota+2015+prius+product+specs.download
>         allocate 1360 to the chassis_floor and the rest to the wheels and the body
> ```

Please change "1360" to be "1360 lbs."

---

*[drake/automotive/models/prius/prius.sdf, line 13 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVliJYCr_i1shyNT3O7:-KVliJYCr_i1shyNT3O8:b4ygtdl) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L13)):*
> ```
>         <!-- curb weight 3,072 lbs. (~1400 kg) based on 
>         https://pressroom.toyota.com/releases/toyota+2015+prius+product+specs.download
>         allocate 1360 to the chassis_floor and the rest to the wheels and the body
> ```

Was the split between the chassis floor and the rest of the vehicle selected arbitrarily? Please update this comment explaining how the distribution was selected. It's OK if the comment says the split was "arbitrarily selected". Just include a `TODO` saying that the current distribution should be replaced with a more accurate one in the future.

---

*[drake/automotive/models/prius/prius.sdf, line 16 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVlj5MgvGNX4Z8PJ8bX:-KVlj5MgvGNX4Z8PJ8bY:b-aq85hl) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L16)):*
> ```
>         -->
>         <mass>1360</mass>
>         <!-- the inertia is calculated by assuming the chassis_floor link as a
> ```

Same comment here about proper punctuation.

---

*[drake/automotive/models/prius/prius.sdf, line 16 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVljBTq4lVvI9h_oY-j:-KVljBTq4lVvI9h_oY-k:b-obl9wt) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L16)):*
> ```
>         -->
>         <mass>1360</mass>
>         <!-- the inertia is calculated by assuming the chassis_floor link as a
> ```

"as" should be "is"

---

*[drake/automotive/models/prius/prius.sdf, line 17 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVljDkJ1ZPjT-MwDX0n:-KVljDkJ1ZPjT-MwDX0o:b-ipqpvz) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L17)):*
> ```
>         <mass>1360</mass>
>         <!-- the inertia is calculated by assuming the chassis_floor link as a
>         cuboid, and using fomula from https://en.wikipedia.org/wiki/List_of_moments_of_inertia
> ```

Typo "fomula"

---

*[drake/automotive/models/prius/prius.sdf, line 17 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVljGJvRg-zXRPnBolP:-KVljGJvRg-zXRPnBolQ:b9n0s9c) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L17)):*
> ```
>         <mass>1360</mass>
>         <!-- the inertia is calculated by assuming the chassis_floor link as a
>         cuboid, and using fomula from https://en.wikipedia.org/wiki/List_of_moments_of_inertia
> ```

Missing "the" after "using"

---

*[drake/automotive/models/prius/prius.urdf, line 14 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVljfyM2o6xuFK7O5uG:-KVljfyM2o6xuFK7O5uH:b-ivngej) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.urdf#L14)):*
> ```XML
>     <inertial>
>       <!-- curb weight 3,072 lbs. (~1400 kg) based on 
>       https://pressroom.toyota.com/releases/toyota+2015+prius+product+specs.download
> ```

Same comment here about having proper punctuation and grammar. You can copy / paste it from `prius.sdf` once its comment is fixed.

---

*[drake/automotive/models/prius/prius_with_lidar.sdf, line 13 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVljpt67rQr8xq29WEh:-KVljpt67rQr8xq29WEi:b-ivngej) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius_with_lidar.sdf#L13)):*
> ```
>         <!-- curb weight 3,072 lbs. (~1400 kg) based on 
>         https://pressroom.toyota.com/releases/toyota+2015+prius+product+specs.download
>         allocate 1360 to the chassis_floor and the rest to the wheels and the body
> ```

Same comment here about having proper punctuation and grammar. You can copy / paste it from `prius.sdf` once its comment is fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-:-KVljtBkf_M6wa-glLOM:b20whe1)*
<!-- Sent from Reviewable.io -->

shensquared commented:
I addressed the new comments, PTAL. Thanks. 

---

Review status: 2 of 6 files reviewed at latest revision, 12 unresolved discussions.

---

*[drake/automotive/car_simulation.cc, line 196 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVlhmXx55n_MsoXus4X:-KVz82OytgeDV3eJK69D:b-7rl0jd) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/car_simulation.cc#L196)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> BTW: nit: missing spaces before and after "/".
</details>
done.

---

*[drake/automotive/car_simulation.cc, line 201 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVlhrf7ftANSEzMWxR3:-KVz8FlVwnheMq2Im9DL:bj8bld6) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/car_simulation.cc#L201)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> BTW: nit: missing spaces before and after "/".
</details>
same as above.

---

*[drake/automotive/models/prius/prius.sdf, line 391 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjj-EOIrzSBx0xrd6T:-KW-6CUsUV_Mmm606uvX:b-5maq7r) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/models/prius/prius.sdf#L391)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Thanks!
</details>
no problem.

---

*[drake/automotive/models/prius/prius.sdf, line 407 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVjkgKTxGRECuZe2mvr:-KVz9LNma5D0KU52C1gT:b-7rl0jd) ([raw file](https://github.com/robotlocomotion/drake/blob/71682c8989b23479dc1b80a2e7a339f31c29b336/drake/automotive/models/prius/prius.sdf#L407)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> OK. Please add a `TODO` comment saying this mass value needs to be verified / fixed.
</details>
done.

---

*[drake/automotive/models/prius/prius.sdf, line 11 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVlijrs2t39l3C8yYLx:-KVz8sC23PqzwS9saSNs:b-hfh5u2) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L11)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> This should be a complete sentence with proper punctuation per style guide.
> 
> ```
> The 2016 Toyota Prius has a curb weight of 3,072 lbs. (~1,400 kg) based on the PDF
> linked to below.
> 
> https://pressroom.toyota.com/releases/toyota+2015+prius+product+specs.download
> ```
</details>
thanks, done. 

---

*[drake/automotive/models/prius/prius.sdf, line 13 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVli9Xw1VnutUrxUIsm:-KVz8wrZbOQuWiO4hJXq:b-wfd32r) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L13)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Please change "1360" to be "1360 lbs."
</details>
it actually is kg. it's the value assigned to the urdf link.

---

*[drake/automotive/models/prius/prius.sdf, line 13 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVliJYCr_i1shyNT3O7:-KVz96OdEb0IncVr2Z4s:b-ae1o0p) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L13)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Was the split between the chassis floor and the rest of the vehicle selected arbitrarily? Please update this comment explaining how the distribution was selected. It's OK if the comment says the split was "arbitrarily selected". Just include a `TODO` saying that the current distribution should be replaced with a more accurate one in the future.
</details>
yes, it's sort of arbitrary. added comment. 

---

*[drake/automotive/models/prius/prius.sdf, line 16 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVlj5MgvGNX4Z8PJ8bX:-KVz9D282YVxgHy7O_R3:b-7rl0jd) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L16)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Same comment here about proper punctuation.
</details>
done.

---

*[drake/automotive/models/prius/prius.sdf, line 16 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVljBTq4lVvI9h_oY-j:-KVz9GSDh0WPXQpfpvPk:b-rnrapz) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L16)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> "as" should be "is"
</details>
done. 

---

*[drake/automotive/models/prius/prius.sdf, line 17 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVljDkJ1ZPjT-MwDX0n:-KVz9G5iJjWtqsnHa41p:b-3ij9pb) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L17)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Typo "fomula"
</details>
sorry about this, done. 

---

*[drake/automotive/models/prius/prius.sdf, line 17 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVljGJvRg-zXRPnBolP:-KVz9JJIFcvQVVeNtabg:b-3ij9pb) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.sdf#L17)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Missing "the" after "using"
</details>
sorry about this, done. 

---

*[drake/automotive/models/prius/prius.urdf, line 14 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVljfyM2o6xuFK7O5uG:-KW-4uqYdYUnP5xI_XUI:b-srv44q) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius.urdf#L14)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Same comment here about having proper punctuation and grammar. You can copy / paste it from `prius.sdf` once its comment is fixed.
</details>
same changes as `prius.sdf`.

---

*[drake/automotive/models/prius/prius_with_lidar.sdf, line 13 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KVljpt67rQr8xq29WEh:-KW-5-BieLoGq8A-8D4Q:b-srv44q) ([raw file](https://github.com/robotlocomotion/drake/blob/299a258c177b7331f440c2155cd2f24551ca8501/drake/automotive/models/prius/prius_with_lidar.sdf#L13)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> Same comment here about having proper punctuation and grammar. You can copy / paste it from `prius.sdf` once its comment is fixed.
</details>
same changes as `prius.sdf`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-:-KW-6GPbFcozFfnmQL3f:bx8k40s)*
<!-- Sent from Reviewable.io -->

liangfok commented:
I completed another round of review.

---

Reviewed 1 of 2 files at r3, 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[drake/automotive/models/prius/prius.sdf, line 14 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW-B3f6IVw6cgrZkcyT:-KW-B3f8FKja63FZwNfM:b3b6rew) ([raw file](https://github.com/robotlocomotion/drake/blob/4456c544cdf7164b71f7b2fef84129e36fa143f4/drake/automotive/models/prius/prius.sdf#L14)):*
> ```
>         https://pressroom.toyota.com/releases/toyota+2015+prius+product+specs.download -->
> 
>         <!-- Our simplified model as of now assumes that only six links have non-zero weight: the chassis_floor, the body, and the four wheels. The chassis_floor's share of the curb weight is set to 1,360 kg, and the only justification for this otherwise arbitrary value is to bring the center of mass reasonable low. -->
> ```

BTW, if possible, please hard-wrap this comment to be 80 characters wide.

---

*[drake/automotive/models/prius/prius.sdf, line 18 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW-BBq5ocAEyeHEuaeU:-KW-BBq5ocAEyeHEuaeV:b-a95ft3) ([raw file](https://github.com/robotlocomotion/drake/blob/4456c544cdf7164b71f7b2fef84129e36fa143f4/drake/automotive/models/prius/prius.sdf#L18)):*
> ```
>         <!-- TODO(shensquared): Refine the weight distribution when the model is of higher fidelity. -->
>         <mass>1360</mass>
>         <!-- the inertia is calculated by assuming the chassis_floor link is a cuboid, and then using the formula from https://en.wikipedia.org/wiki/List_of_moments_of_inertia
> ```

BTW: nit: "the" should be "The". Also, missing period at end of sentence.

---

*[drake/automotive/models/prius/prius.sdf, line 408 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW-BNJdirfYVPTMTzNf:-KW-BNJdirfYVPTMTzNg:bym71t4) ([raw file](https://github.com/robotlocomotion/drake/blob/4456c544cdf7164b71f7b2fef84129e36fa143f4/drake/automotive/models/prius/prius.sdf#L408)):*
> ```
>         <pose>0 0 0 0 0 0</pose>
>         <mass>54.2794</mass>
>         <!-- TODO(shensquared): Work with Steve to verify the inertia values below are reasonable -->
> ```

BTW: nit: missing period at end of sentence.

---

*[drake/automotive/models/prius/prius.urdf, line 22 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW-BdOcsnz0IxnsmURU:-KW-BdOcsnz0IxnsmURV:bytvt4s) ([raw file](https://github.com/robotlocomotion/drake/blob/4456c544cdf7164b71f7b2fef84129e36fa143f4/drake/automotive/models/prius/prius.urdf#L22)):*
> ```XML
>       <mass value="1360"/>
>       <origin rpy="0 0 0" xyz="0 0 0" />
>       <!-- the inertia is calculated by assuming the chassis_floor link is a cuboid, and then using the formula from https://en.wikipedia.org/wiki/List_of_moments_of_inertia
> ```

BTW: same comment as in `prius.sdf`.

---

*[drake/automotive/models/prius/prius.urdf, line 297 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW-BWhwX85Jkwe1jJyH:-KW-BWhwX85Jkwe1jJyI:bytvt4s) ([raw file](https://github.com/robotlocomotion/drake/blob/4456c544cdf7164b71f7b2fef84129e36fa143f4/drake/automotive/models/prius/prius.urdf#L297)):*
> ```XML
>       <mass value="54.2794"/>
>       <origin rpy="0 0 0" xyz="0 0 0"/>
>       <!-- TODO(shensquared): Work with Steve to verify the inertia values below are reasonable -->
> ```

BTW: same comment as in `prius.sdf`.

---

*[drake/automotive/models/prius/prius_with_lidar.sdf, line 18 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW-BewzhvGku6YbaEAr:-KW-Bex-39ba3Lu0cOp4:bytvt4s) ([raw file](https://github.com/robotlocomotion/drake/blob/4456c544cdf7164b71f7b2fef84129e36fa143f4/drake/automotive/models/prius/prius_with_lidar.sdf#L18)):*
> ```
>         <!-- TODO(shensquared): Refine the weight distribution when the model is of higher fidelity. -->
>         <mass>1360</mass>
>         <!-- the inertia is calculated by assuming the chassis_floor link is a cuboid, and then using the formula from https://en.wikipedia.org/wiki/List_of_moments_of_inertia
> ```

BTW: same comment as in `prius.sdf`.

---

*[drake/automotive/models/prius/prius_with_lidar.sdf, line 408 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW-BYiWWERZMB0BdApB:-KW-BYiWWERZMB0BdApC:bytvt4s) ([raw file](https://github.com/robotlocomotion/drake/blob/4456c544cdf7164b71f7b2fef84129e36fa143f4/drake/automotive/models/prius/prius_with_lidar.sdf#L408)):*
> ```
>         <pose>0 0 0 0 0 0</pose>
>         <mass>54.2794</mass>
>         <!-- TODO(shensquared): Work with Steve to verify the inertia values below are reasonable -->
> ```

BTW: same comment as in `prius.sdf`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-:-KW-BiCXI02pOPswDkVS:bl1vavh)*
<!-- Sent from Reviewable.io -->

liangfok commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 3 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-:-KW-NdPas7Wj8GWpUH4r:bnfp4nl)*
<!-- Sent from Reviewable.io -->

liangfok commented:
+@sherm1 for platform review per rotation. Thanks.

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-:-KW-OnCF3cioChsjfW5h:b-oxq2sg)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4023)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Platform: looks good except for duplication. PTAL and see if you can figure out a way for the detailed modeling information to appear only once.

---

Reviewed 1 of 6 files at r1, 1 of 5 files at r2, 1 of 2 files at r3, 3 of 3 files at r6.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[drake/automotive/models/prius/prius.sdf, line 26 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW1_5R1PMy0Xf8n3syT:-KW1_5R27L2fY6NDxS3L:bx7qyb7) ([raw file](https://github.com/robotlocomotion/drake/blob/c520491081bcc9a65d4ab788472ab4bd43c64c94/drake/automotive/models/prius/prius.sdf#L26)):*
> ```
>         <mass>1360</mass>
>       <!-- The inertia is calculated by assuming the chassis_floor link is a
>       cuboid, and then using the formula from
> ```

BTW, please give the dimensions of the cuboid.

---

*[drake/automotive/models/prius/prius.urdf, line 13 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW1_Z5jyFKIYkR3ZrcD:-KW1_Z5jyFKIYkR3ZrcE:bhxhd6h) ([raw file](https://github.com/robotlocomotion/drake/blob/c520491081bcc9a65d4ab788472ab4bd43c64c94/drake/automotive/models/prius/prius.urdf#L13)):*
> ```XML
>   <link name="chassis_floor">
>     <inertial>
>       <!--  The 2015 Prius has a curb weight of 3,072 lbs. (~1,400 kg)
> ```

Is it necessary to have both an `.sdf` and a `.urdf`? This seems like an unreasonable amount of duplication which will almost inevitably lead to divergence.

---

*[drake/automotive/models/prius/prius.urdf, line 30 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW1_KPp1kY0frNE4hJV:-KW1_KPp1kY0frNE4hJW:b-3t817z) ([raw file](https://github.com/robotlocomotion/drake/blob/c520491081bcc9a65d4ab788472ab4bd43c64c94/drake/automotive/models/prius/prius.urdf#L30)):*
> ```XML
>       <origin rpy="0 0 0" xyz="0 0 0" />
>       <!-- The inertia is calculated by assuming the chassis_floor link is a
>       cuboid, and then using the formula from
> ```

BTW, please give dimensions of cuboid.

---

*[drake/automotive/models/prius/prius_with_lidar.sdf, line 11 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW1_rIeESetsVdaIrrz:-KW1_rIeESetsVdaIrs-:b8wb4sf) ([raw file](https://github.com/robotlocomotion/drake/blob/c520491081bcc9a65d4ab788472ab4bd43c64c94/drake/automotive/models/prius/prius_with_lidar.sdf#L11)):*
> ```
>       <inertial>
>        <pose>0 0 0 0 0 0</pose>
>       <!--  The 2015 Prius has a curb weight of 3,072 lbs. (~1,400 kg)
> ```

We are now up to three copies of this same detailed code! Is there a way to reduce this duplication?

---

*[drake/automotive/models/prius/prius_with_lidar.sdf, line 26 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW1a1YLmCTYZhYKY8l9:-KW1a1YLmCTYZhYKY8lA:b-fdevii) ([raw file](https://github.com/robotlocomotion/drake/blob/c520491081bcc9a65d4ab788472ab4bd43c64c94/drake/automotive/models/prius/prius_with_lidar.sdf#L26)):*
> ```
>         <mass>1360</mass>
>         <!-- The inertia is calculated by assuming the chassis_floor link is a
>         cuboid, and then using the formula from
> ```

BTW, give dimensions of cuboid -- but only once!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-:-KW1a9936-PSRu6E4KYk:baobrqe)*
<!-- Sent from Reviewable.io -->

liangfok commented:
I added some comments about how to avoid duplication, though I think it should be punted to a future PR. See issue #4074.

---

Reviewed 3 of 3 files at r7.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[drake/automotive/models/prius/prius.urdf, line 13 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW1_Z5jyFKIYkR3ZrcD:-KW3NSDQWbRW564Gxbev:b-o27g1c) ([raw file](https://github.com/robotlocomotion/drake/blob/c520491081bcc9a65d4ab788472ab4bd43c64c94/drake/automotive/models/prius/prius.urdf#L13)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary>
> Is it necessary to have both an `.sdf` and a `.urdf`? This seems like an unreasonable amount of duplication which will almost inevitably lead to divergence.
</details>
IMHO, there's currently no way to avoid having both URDF and SDF in the tree.

There is a semi-official way to convert from URDF to SDF:

```
gz sdf -p foo.urdf > foo.sdf
```

It's not perfect though and typically requires hand-tuning. Also, since URDF is effectively deprecated in favor of SDF, it would be strange to remove the SDF.

There is an unofficial converter from SDF to URDF:

https://github.com/andreasBihlmaier/pysdf

From my experience, it requires even more hand tuning of the results, meaning we can't remove the URDF and automatically generate it based on the SDF.

Keeping the URDF is useful because it's currently the only format that can be loaded into ROS' `robot_description` parameter, which is used by various applications like [RViz](http://wiki.ros.org/rviz), [TF](http://wiki.ros.org/tf/), and [joint_state_publisher](http://wiki.ros.org/joint_state_publisher).

---

*[drake/automotive/models/prius/prius_with_lidar.sdf, line 11 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW1_rIeESetsVdaIrrz:-KW3PtDRpcAKZPW3ajQf:b-jldoe0) ([raw file](https://github.com/robotlocomotion/drake/blob/c520491081bcc9a65d4ab788472ab4bd43c64c94/drake/automotive/models/prius/prius_with_lidar.sdf#L11)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary>
> We are now up to three copies of this same detailed code! Is there a way to reduce this duplication?
</details>
In URDF, [xacro](http://wiki.ros.org/xacro) can be used to avoid duplication.

In SDF, there appears to be embedded ruby:

http://answers.gazebosim.org/question/7213/parameterizing-elements-of-an-sdf-model/

I think applying embedded ruby in this PR may be too much and have thus created #4074 to track this improvement.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-:-KW3RZXVLU77hcOVEUuZ:bokpyil)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> -- I agree sdf/urdf duplication is a bigger issue to be handled elsewhere.

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/automotive/models/prius/prius.urdf, line 13 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-KW1_Z5jyFKIYkR3ZrcD:-KW3p9UsNlxfbeLnLgcF:b-d38qhk) ([raw file](https://github.com/robotlocomotion/drake/blob/c520491081bcc9a65d4ab788472ab4bd43c64c94/drake/automotive/models/prius/prius.urdf#L13)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary>
> IMHO, there's currently no way to avoid having both URDF and SDF in the tree.
> 
> There is a semi-official way to convert from URDF to SDF:
> 
> ```
> gz sdf -p foo.urdf > foo.sdf
> ```
> 
> It's not perfect though and typically requires hand-tuning. Also, since URDF is effectively deprecated in favor of SDF, it would be strange to remove the SDF.
> 
> There is an unofficial converter from SDF to URDF:
> 
> https://github.com/andreasBihlmaier/pysdf
> 
> From my experience, it requires even more hand tuning of the results, meaning we can't remove the URDF and automatically generate it based on the SDF.
> 
> Keeping the URDF is useful because it's currently the only format that can be loaded into ROS' `robot_description` parameter, which is used by various applications like [RViz](http://wiki.ros.org/rviz), [TF](http://wiki.ros.org/tf/), and [joint_state_publisher](http://wiki.ros.org/joint_state_publisher).
</details>
I agree this can be addressed elsewhere.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-:-KW3qDyWUZtWkJo15ZL0:byhnyuv)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Thanks. I will go ahead and merge this branch.

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/4023#-:-KW4FAVs1wwHS_jYHuVH:b50euxn)*
<!-- Sent from Reviewable.io -->
